TextureHolder.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <SFML/Graphics.hpp>
+#include <map>
+#include <memory>
+
+class TextureHolder
+{
+public:
+	TextureHolder();

```

Comment:
Technically, the constructor should be private!

Another way to do this would be to have

```
// the header file is just this

class TextureHolder {
public:
  static ... GetTexture(...);
}

//-------
// and in the cpp file

// There is a class declaration and definition in the cpp file! This one does the actual work and it hides its methods from public view that would be exposed in TextureHolder.h
class TextureHolderImpl {
public:
  std::map<std::string, sf::Texture> m_Textures;
} s_Instance; // This is now the singleton, it is magically created when the program starts

sf::Texture& TextureHolder(...) {
  s_Instance ...
}
```